### PR TITLE
feat(transport): static native `io_uring` support

### PR DESCRIPTION
**Pull request by sgammon**
Ready for review [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=elide-dev&utm_campaign=badge_info)

## Summary

Adds static JNI support for Linux's `io_uring` interface, through Netty's incubator transport.

## Changelog

- chore: `io_uring` native lib unpack
- chore: dep adjustments to exclude existing natives
- chore: prepare direct static exports
- chore: add sources for netty's io_uring layer

### CODE REVIEW

File: packages/graalvm/src/main/kotlin/elide/runtime/feature/engine/NativeSQLiteFeature.kt [hunk 1]
```diff
@@ -40,10 +40,10 @@ import elide.runtime.feature.NativeLibraryFeature.UnpackedNative
...
 
   override fun nativeLibs(access: BeforeAnalysisAccess): List<NativeLibInfo> = if (STATIC_JNI) listOfNotNull(
-    nativeLibrary(singular = libraryNamed(
-      "sqlitejdbc",
-      "org.sqlite.core.NativeDB",
-    )),
+    nativeLibrary(singular = libraryNamed("sqlitejdbc", "org.sqlite.core.NativeDB", deps = (
+      // on linux, we need to additionally link against the `math` library, which is at `libm`
+      if (Platform.includedIn(Platform.LINUX::class.java) && STATIC_JNI) listOf("m") else emptyList()
+    )))
```
Comment (sgammon) [inline packages/graalvm/src/main/kotlin/elide/runtime/feature/engine/NativeSQLiteFeature.kt:46]: sqlite fix slipped in, it just adds `-lm` on linux (links against libmath)

File: packages/graalvm/src/main/kotlin/elide/runtime/feature/engine/NativeTransportFeature.kt [hunk 2]
```diff
@@ -120,7 +120,14 @@ import elide.runtime.feature.NativeLibraryFeature.UnpackedNative
...
         renameTo = { "libnetty_transport_native_epoll.${it.substringAfterLast(".")}" },
-      )
+      ).plus(access.unpackLibrary(
+        "transport-uring",
+        "netty_transport_native_io_uring",
+        "x86-64",
+        "META-INF/native/libtransport-uring.a",  // @TODO: gradle doesn't support arm64 linux
+        "META-INF/native/libtransport-uring.so",
+        renameTo = { "libnetty_transport_native_io_uring.${it.substringAfterLast(".")}" },
+      ))
```
Comment (sgammon) [inline packages/graalvm/src/main/kotlin/elide/runtime/feature/engine/NativeTransportFeature.kt:130]: `epoll` and `io_uring` are both unpacked and loaded on linux

File: packages/runtime/build.gradle.kts [hunk 3]
```diff
@@ -583,8 +585,8 @@ val deprecatedNativeArgs = listOf(
 )
 
 val enabledFeatures = listOfNotNull(
-  "com.sun.jna.SubstrateStaticJNA",
   "elide.tool.feature.ToolingUmbrellaFeature",
+  onlyIf(HostManager.hostIsMac, "com.sun.jna.SubstrateStaticJNA"),
```
Comment (sgammon) [inline packages/runtime/build.gradle.kts:589]: static JNA is limited to mac until we fix the libs upstream

File: packages/runtime/src/main/resources/META-INF/native-image/reflect-config.json [hunk 4]
```diff
@@ -7494,6 +7494,13 @@
   "allDeclaredConstructors":true,
   "allPublicConstructors":true
 },
+{
+  "name":"com.sun.jna.Native",
+  "allDeclaredMethods":true,
+  "allPublicMethods":true,
+  "allDeclaredConstructors":true,
+  "allPublicConstructors":true
+},
```
Comment (sgammon) [inline packages/runtime/src/main/resources/META-INF/native-image/reflect-config.json:7503]: same here, missing some reflection config upstream

File: packages/transport/transport-uring/src/main/cpp/netty_io_uring_linuxsocket.c [hunk 5]
```diff
@@ -0,0 +1,966 @@
...
+// private static native void joinSsmGroup(int fd, boolean ipv6, byte[] group, byte[] interfaceAddress,
+//                                         int scopeId, int interfaceIndex, byte[] source) throws IOException;
+// private static native void leaveGroup(int fd, boolean ipv6, byte[] group, byte[] interfaceAddress,
+//                                       int scopeId, int interfaceIndex) throws IOException;
+// private static native void leaveSsmGroup(int fd, boolean ipv6, byte[] group, byte[] interfaceAddress,
+//                                           int scopeId, int interfaceIndex, byte[] source) throws IOException;
+
+JNIEXPORT void Java_io_netty_incubator_channel_uring_LinuxSocket_joinGroup(JNIEnv* env, jclass clazz, jint fd, jboolean ipv6, jbyteArray groupAddress, jbyteArray interfaceAddress, jint scopeId, jint interfaceIndex) {
+    netty_io_uring_linuxsocket_joinGroup(env, clazz, fd, ipv6, groupAddress, interfaceAddress, scopeId, interfaceIndex);
+};
```
Comment (sgammon) [inline packages/transport/transport-uring/src/main/cpp/netty_io_uring_linuxsocket.c:761]: exports for our static direct layer

**Review by darvld — APPROVED** [review]

**Pull request merged**